celery/app/task.py

```diff
@@ -27,6 +27,14 @@
 from .registry import _unpickle_task_v2
 from .utils import appstr
 
+class RetryEagerException(Retry):
```

Comment:
Why do we need a different type?
Couldn't we include the eagerness information inside the Retry exception?

Reply:
It should be easy to reuse [Retry](https://github.com/celery/celery/blob/92ced78168b1ad92268af039827a9df957fc90af/celery/exceptions.py#L140):

```python
def __init__(self, message=None, exc=None, when=None, **kwargs):
    self.signature = kwargs.get('signature')
    self.eager = kwargs.get('eager', False)
```